platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -58,44 +76,75 @@
     @Nullable
     private final Schema actualSchema;
     private final String ssTablePath;
+    private long rowCount;
 
     public boolean lazyLoading;
     public final CFMetaData metadata;
     private final SSTableReader ssTableReader;
 
-    static{
-        // Initializing DatabaseDescriptor that contains Cassandra's general configuration
+    private final PartitionColumns columnDefinitions;
+    private final IPartitioner partitioner;
+    private final List<CassandraTokenRange> tokenRanges;
+    private final String localEndpoint;
+
+    static {
+        // Initializing Cassandra's general configuration
         DatabaseDescriptor.clientInitialization(false);
     }
 
-    public CassandraSSTableLoader(String ssTablePath, boolean lazyLoading) {
+    public CassandraSSTableLoader(String ssTablePath, List<CassandraTokenRange> tokenRanges, String localEndpoint,
+            boolean lazyLoading) {
         super(ssTablePath);
         this.ssTablePath = ssTablePath;
+        this.tokenRanges = tokenRanges;
+        this.localEndpoint = localEndpoint;
         this.lazyLoading = lazyLoading;
 
         try {
             Descriptor descriptor = Descriptor.fromFilename(this.ssTablePath);
+
+            this.partitioner = FBUtilities.newPartitioner(descriptor);
             this.metadata = this.getSSTableMetadata(descriptor);
+            this.columnDefinitions = this.metadata.partitionColumns();
             this.ssTableReader = SSTableReader.openNoValidation(descriptor, this.metadata);
-            // Get the schema of the current SSTable
-            this.actualSchema = this.getSchema(this.metadata);
+            this.actualSchema = this.convertSchema();
+            this.computeRowCount();
+        } catch (Exception e) {
+            HillviewLogger.instance.error("Failed initializing Metadata and SStable partitions", "{0}",
+                    this.ssTablePath);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
```

Comment:
Please add a comment that this function is only used for testing.
The function name should also reflect that: getTestTableLoader?